crates/rpc/rpc-eth-types/src/cache/mod.rs

```diff
@@ -580,6 +545,78 @@ impl<B: Block, R: Clone> ChainChange<B, R> {
     }
 }
 
+/// Identifier for the caches.
+#[derive(Copy, Clone, Debug)]
+enum CacheKind {
+    Block,
+    Receipt,
+    Header,
+}
+
+/// Drop aware sender struct that ensures a response is always emitted even if the db task panics
+/// before a result could be sent.
+///
+/// This type wraps a sender and in case the sender is still present on drop emit an error response.
+#[derive(Debug)]
+struct ActionSender<B: Block, R: Send + Sync> {
+    kind: CacheKind,
+    blockhash: B256,
+    tx: Option<UnboundedSender<CacheAction<B, R>>>,
+}
+
+impl<R: Send + Sync, B: Block> ActionSender<B, R> {
+    const fn new(
+        kind: CacheKind,
+        blockhash: B256,
+        tx: Option<UnboundedSender<CacheAction<B, R>>>,
```

Comment:
tx in constructor is always be Some so shouldnt take in option

Reply:
yes but I need to know on drop whether we've already sent a response.

this should behaves like a oneshot channel here

Reply:
it should be stored as an option, just not took in as argument

Reply:
fair